app/frontend/src/features/search/SearchPage.tsx

```diff
@@ -1,51 +1,181 @@
+import smoothscroll from "smoothscroll-polyfill";
+
+import { selectedUserZoom } from "./constants";
+import SearchResultsList from "./SearchResultsList";
+import { addUsersToMap } from "./users";
+
+smoothscroll.polyfill();
+
+const useStyles = makeStyles((theme) => ({
+  container: {
+    display: "flex",
+    alignContent: "stretch",
+    flexDirection: "column-reverse",
+    height: `calc(100vh - ${theme.shape.navPaddingMobile})`,
+    marginTop: theme.shape.navPaddingMobile,
+    [theme.breakpoints.up("md")]: {
+      flexDirection: "row",
+      height: `calc(100vh - ${theme.shape.navPaddingDesktop})`,
+      marginTop: theme.shape.navPaddingDesktop,
+    },
+  },
+  mapContainer: {
+    flexGrow: 1,
+    position: "relative",
+  },
+  mapResults: {
+    height: "14rem",
+    zIndex: 3,
+    overflow: "hidden",
+    [theme.breakpoints.up("md")]: {
+      height: "auto",
+      width: "35%",
+      overflow: "auto",
+    },
+  },
+  searchMobile: {
+    margin: theme.spacing(0, 2),
+    position: "absolute",
+    top: theme.spacing(1),
+    left: theme.spacing(1),
+    right: 36,
+    paddingBottom: theme.spacing(1),
+    paddingLeft: theme.spacing(1),
+    paddingRight: theme.spacing(1),
+    backgroundColor: "rgba(255, 255, 255, 0.8)",
+    borderRadius: theme.shape.borderRadius,
+  },
+}));
 
 export default function SearchPage() {
-  const [loading, setLoading] = useState(false);
-  const [error, setError] = useState("");
-  const [results, setResults] = useState([] as User.AsObject[]);
+  const classes = useStyles();
+  const theme = useTheme();
 
-  const { query } = useParams<SearchQuery>();
+  const map = useRef<MaplibreMap>();
+  const [selectedResult, setSelectedResult] = useState<number | undefined>(
+    undefined
+  );
+
+  const { query } = useParams<{ query?: string }>();
   useEffect(() => {
-    if (!query) return;
-    (async () => {
-      setLoading(true);
-      try {
-        setResults(await service.search.search(query));
-      } catch (e) {
-        setError(e.message);
+    setTimeout(
+      () => map.current?.resize(),
+      theme.transitions.duration.standard
+    );
+  }, [query, theme.transitions.duration.standard]);
+
+  const history = useHistory();
+  /*const location = useLocation();
+
+  const handlePlaceClick = (ev: any) => {
+    const properties = ev.features[0].properties as {
+      id: number;
+      slug: string;
+    };
+    history.push(routeToPlace(properties.id, properties.slug), location.state);
+  };
+
+  const handleGuideClick = (ev: any) => {
+    const properties = ev.features[0].properties as {
+      id: number;
+      slug: string;
+    };
+    history.push(routeToGuide(properties.id, properties.slug), location.state);
+  };*/
+
+  const flyToUser = (user: Pick<User.AsObject, "lng" | "lat">) => {
+    map.current?.stop();
+    map.current?.flyTo({
+      center: [user.lng, user.lat],
+      zoom: selectedUserZoom,
+    });
+  };
+
+  const handleMapUserClick = (ev: any) => {
```

Comment:
Not sure where best to put this but seems related to this function: is it intentional that clicking a user on the map only highlights them in the search result rather than taking you directly to the user's profile (like how the old map page works)?

Reply:
Yes, if you click them again, it goes to their profile. Taking suggestions for a better idea! (But there needs to be a way to highlight them on the list/map first)

Reply:
I think that bit where you have to click twice in the search result is probably fine (for me at least). What I meant was on the map itself - I was expecting clicking the marker to also go to the user's profile when it's only for one person but it doesn't do that anymore, which the old map did.

Maybe we should pop up something when the marker is clicked and the user summary (but probably a shorter one than one shown in search result?) is shown in the pop up? Then that could be clickable and takes the user to a profile. Shamelessly lifting the idea from other apps with map. Not sure how tricky that might be to do though, so maybe one for the next PR?

Reply:
Yeah I think save that for later.